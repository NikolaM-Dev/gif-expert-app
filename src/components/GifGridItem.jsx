import PropTypes from 'prop-types';

const GifGridItem = ({ gif }) => {
  return (
    <figure className="card animate__animated animate__fadeIn">
      <img src={gif.url} alt={gif.title} />
      <figcaption>{gif.title}</figcaption>
    </figure>
  );
};

GifGridItem.propTypes = {
  gif: PropTypes.object.isRequired,
};

export default GifGridItem;
