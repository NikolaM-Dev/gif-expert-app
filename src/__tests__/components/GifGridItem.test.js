import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import GifGridItem from 'components/GifGridItem';

describe('Tests on <GifGridItem />', () => {
  const gif = {
    title: 'Gif title',
    url: 'https://gifUrl.com',
  };

  const wrapper = shallow(<GifGridItem gif={gif} />);

  it('should show <GifGridItem /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a figcaption with the gif.title', () => {
    const figcaptionText = wrapper.find('figcaption').text().trim();

    expect(figcaptionText).toBe(gif.title);
  });

  it('should have a img with the url and alt from props', () => {
    const { src, alt } = wrapper.find('img').props();

    expect(src).toBe(gif.url);
    expect(alt).toBe(gif.title);
  });

  it('should have the className animate__fadeIn', () => {
    const figure = wrapper.find('figure');
    const className = figure.prop('className');

    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
