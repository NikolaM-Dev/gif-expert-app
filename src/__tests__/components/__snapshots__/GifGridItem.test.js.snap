// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> should show <GifGridItem /> correctly 1`] = `
<figure
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Gif title"
    src="https://gifUrl.com"
  />
  <figcaption>
    Gif title
  </figcaption>
</figure>
`;
